mongofiles/mongofiles.go

```diff
@@ -248,6 +261,22 @@ func (mf *MongoFiles) getTargetGFSFiles() ([]*gfsFile, error) {
 		if len(gridFiles) < len(mf.FileNameList) {
 			return nil, fmt.Errorf("requested files not found: %v", mf.FileNameList)
 		}
+	} else if mf.FileNameRegex != "" {
```

Comment:
This method is now quite large and contains effectively no code shared by the three branches of the conditional. I wonder if refactoring is in order here. I see three options:
1. Leave this as-is
2. Separate the three branches into three separate methods
3. Factor this method so the three branches share common code. It looks like each branch determines what the `query` is to argue to `mf.findGFSFiles` and each branch returns a unique error if `len(gridFiles)` is unexpected. Perhaps the three branches can set `query` and some error format string and then common code can run `mf.findGFSFiles(query)` and return an error if something is wrong.

Reply:
I don't want to break this off and fragment the code structure, so I'd personally prefer (3).

I'm interested in thoughts from @huan-Mongo and @tfogo, however!

Reply:
(2) or (3) sounds good. I imagine we'll be able to get this done this week.

Reply:
I'll move forward on (3) then, as long as Huan doesn't have any objections.

Reply:
3 Looks good to me!